zcash_transparent/src/builder.rs

```diff
@@ -323,12 +316,15 @@ impl TransparentAuthorizingContext for Unauthorized {
 }
 
 impl Bundle<Unauthorized> {
-    pub fn apply_signatures(
+    #[cfg_attr(not(feature = "transparent-inputs"), allow(unused_variables))]
+    pub fn apply_signatures<F>(
         self,
-        #[cfg(feature = "transparent-inputs")] mtx: &TransactionData<tx::Unauthorized>,
-        #[cfg(feature = "transparent-inputs")] txid_parts_cache: &TxDigests<Blake2bHash>,
+        calculate_sighash: F,
         signing_set: &TransparentSigningSet,
-    ) -> Result<Bundle<Authorized>, Error> {
+    ) -> Result<Bundle<Authorized>, Error>
+    where
+        F: Fn(&SighashType, usize, &Script, &Script, NonNegativeAmount) -> [u8; 32],
```

Comment:
Nonblocking nit: the type of this closure isn't very informative; could the argument to it be a struct so that the individual parts can be documented more easily? This is effectively how the `SignableInput::Transparent` variant used to work; it's basically just pulling out that variant as a struct up in `zcash_transparent`.